src/cmd/usb-util.js

```diff
@@ -351,17 +420,63 @@ async function forEachUsbDevice(args, func, { dfuMode = false } = {}){
 	const msg = 'Getting device information...';
 	const operation = openUsbDevices(args, { dfuMode });
 	let lastError = null;
+	let outputMsg = [];
 	return spin(operation, msg)
 		.then(usbDevices => {
 			const p = usbDevices.map(usbDevice => {
+				let deviceIsProtected = false;
+				const deviceId = usbDevice.id;
+				const firmwareVersion = usbDevice._fwVer;
 				return Promise.resolve()
-					.then(() => func(usbDevice))
+					.then(async () => {
+						try {
+							const s = await deviceProtectionHelper.getProtectionStatus(usbDevice);
+							deviceIsProtected = s.protected && !s.overridden;
+							if (deviceIsProtected) {
+								if (usbDevice.isInDfuMode) {
+									// We have to put the Protected Device in Service Mode and for this the device should be
+									// put in safe mode first, disable DP, and then put the device back in DFU mode.
+									// Instead, ask the user to start the device in normal mode and if the device requires to be
+									// in dfu mode for the CLI operation, then CLI will take care of putting in in dfu mode
+									// after putting the device in Service Mode
+									throw new Error('Cannot run this command on a Protected Device in DFU mode. Please exit DFU mode and try again.');
+								}
+								await deviceProtectionHelper.disableDeviceProtection(usbDevice);
+							}
+						} catch (err) {
+							if (err.message === 'Not supported' || err.message === 'Request Error') {
+								// Device Protection is not supported on certain platforms and versions.
+								// It means that the device is not protected.
+								// XXX: I would not expect the getProtectionStatus to work and
+								// disableDeviceProtection to throw this error.
+							}
+						}
+						return func(usbDevice);
+					})
 					.catch(e => lastError = e)
-					.finally(() => usbDevice.close());
+					.finally(async () => {
+						if (deviceIsProtected) {
+							// if device goes into DFU mode, we need to reopen it
+							usbDevice = await reopenDevice({ id : deviceId });
+							// XXX: Cannot turn off Service Mode with device-os < 6.1.3 (TBD) if the device is in DFU mode
+							if (usbDevice.isInDfuMode && semver.lt(firmwareVersion, '6.1.3')) {
+								outputMsg.push(`Your device may be in Service Mode. Re-enable Device Protection by running ${chalk.yellow('particle device-protection enable')}`);
+							} else {
+								// TODO (discuss): Add a retry mechanism if this fails
+								await deviceProtectionHelper.turnOffServiceMode(usbDevice);
+							}
+						}
+						if (usbDevice && usbDevice.isOpen){
+							await usbDevice.close();
+						}
+					});
```

Comment:
```
if (usbDevice.isInDfuMode && semver.lt(firmwareVersion, '6.1.3')) {
								outputMsg.push(`Your device may be in Service Mode. Re-enable Device Protection by running ${chalk.yellow('particle device-protection enable')}`);
```

this will cause `particle usb dfu` to have this error as well. I will fix this in usb module or here if you prefer